src/modes/object_to_mode.js

```diff
@@ -41,15 +41,14 @@ module.exports = function(modeObject) {
         // handlers that are not present in the mode
         // to reduce on render calls for functions that
         // have no logic
-        eventKeys.forEach((key) => {
+        eventKeys.forEach(key => {
           const modeHandler = eventMapper[key];
           let selector = () => false;
           if (modeObject[modeHandler]) {
             selector = () => true;
           }
-          this.on(key, selector, wrapper(modeHandler));
+          this.on(key, selector, _throttle(wrapper(modeHandler), 40));
```

Comment:
this is new. This throttles all event handlers. Although our primary problem was related to snap, even without snap performance was mediocre, this improves it.

Reply:
@msimms1123 I think that you said throttled 100 ms in scrum, just making sure you wanted 40 not 100.

Edit: Oh I see, everything is throttled at 40 and snap is throttled at 100?

Reply:
thats right. performance w old snapping was like a 2/10. but even w/o snapping it wasnt great. So I threw an extra throttle on here and it seems to help